tmt/checks/coredump.py

```diff
@@ -0,0 +1,350 @@
+                    ).stdout
+
+                if not crash_info:
+                    logger.debug(f"No crash info available for PID {pid}")
+                    continue
+
+                # Skip if this crash matches any ignore pattern
+                if self.ignore_patterns:
+                    ignored = False
+                    for pattern in self.ignore_patterns:
+                        if pattern.search(crash_info):
+                            logger.debug(
+                                f"Ignoring crash due to pattern '{pattern.pattern}': {crash_info}"
+                                )
+                            ignored = True
+                            break
+                    if ignored:
+                        continue
+
+                # Save the crash info
+                info_file = f"{check_files_path}/dump.{exe}_{sig}_{pid}.txt"
+                guest.execute(
+                    Command("sh", "-c", f"coredumpctl info --no-pager {pid} > {info_file}")
+                    )
+                logger.debug(f"Saved crash info to {info_file}")
+
+                # Try to save the coredump if available
+                if corefile not in ("none", "missing"):
+                    dump_file = f"{check_files_path}/dump.{exe}_{sig}_{pid}.core"
+                    try:
+                        guest.execute(
+                            Command("coredumpctl", "dump", "--no-pager", "-o", dump_file, pid)
+                            )
+                        logger.debug(f"Saved coredump to {dump_file}")
+                    except tmt.utils.RunError as exc:
+                        logger.debug(f"Failed to save coredump for PID {pid}: {exc}")
+                else:
+                    logger.debug(f"Skipping dump for PID {pid}, corefile status: {corefile}")
+
+                # This is a non-ignored crash
+                has_crashes = True
+
+            return has_crashes
+
+        except tmt.utils.RunError as exc:
+            logger.debug(f"Failed to check for crashes: {exc}")
+            return False
+
+    def _check_coredump(
+            self, invocation: "TestInvocation", event: CheckEvent, logger: tmt.log.Logger
+            ) -> ResultOutcome:
+        """Check coredump status and return appropriate result."""
+        # Only configure coredump in before_test
+        if event == CheckEvent.BEFORE_TEST:
+            # Configure coredump
+            if not self._configure_coredump(
+                    invocation.guest, logger, str(invocation.check_files_path)):
+                return ResultOutcome.ERROR
+
+            # Create timestamp directory and store current time for --since filtering
+            invocation.guest.execute(
+                Command(
+                    "sh",
+                    "-c",
+                    f"mkdir -p {DEFAULT_TMP_PATH} && "
+                    f"date '+%Y-%m-%d %H:%M:%S' > {DEFAULT_TMP_PATH}/coredump-timestamp"
```

Comment:
A single, shared, mutable, global state. All tmt invocations sharing the guest - e.g. with `provision -h local` - will change this location without any kind of synchronization. It's better to use a directory that cannot conflict with anyone else, e.g. AVC check uses `ausearch_timestamp_filepath = invocation.check_files_path / AUSEARCH_TIMESTAMP_FILENAME`.

Reply:
Plus the `sh -c`, seems to be present in many commands. `ShellScript` will serve us better, `guest.execute(ShellScript(f"mkdir -p {DEFAULT_TMP_PATH} && date '+%Y-%m-%d %H:%M:%S' > {DEFAULT_TMP_PATH}/coredump-timestamp"))`